include/oaknut/oaknut.hpp

```diff
@@ -163,6 +169,18 @@ class BasicCodeGenerator : public Policy {
         }
     }
 
+    // Convenience function for moving pointers to registers 
+    void MOVP2R(XReg xd, void* addr) {
+        int64_t diff = reinterpret_cast<uint64_t>(addr) - Policy::current_address();
+        if (diff >= -0xF'FFFF && diff <= 0xF'FFFF) {
+            ADR(xd, addr);
+        } else if (diff >= -0xFFFF'FFFF && diff <= 0xFFFF'FFFF) {
```

Comment:
```suggestion
        } else if (diff >= -std::uint64{0xFFFF'FFFF} && diff <= std::uint64{0xFFFF'FFFF}) {
```

`-0xFFFF'FFFF` will get misinterpreted otherwise.